cmd/skopeo/sync.go

```diff
@@ -780,6 +808,10 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) (retErr error) {
 	} else {
 		logrus.Infof("Synced %d images from %d sources", imagesNumber, len(srcRepoList))
 	}
+	// If there were errors during the sync process, return an error
+	if err != nil {
```

Comment:
What does this do? AFAICS `err` is always `nil` on this path (and if it isn’t, that should be detected locally where that happens).

Guessing, this might have been intended to work with the added `defer` handler of `Close` but that’s not how `defer` works; it is not yet called at this point.